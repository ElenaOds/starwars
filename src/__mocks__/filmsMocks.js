export const filmsMock = [
    {id: "1", title: 'A New Hope'},
    {id: "2", title: 'The Empire Strikes Back'},
    {id: "3", title: 'Return of the Jedi'},
    {id: "6", title: '"Revenge of the Sith'},
]
