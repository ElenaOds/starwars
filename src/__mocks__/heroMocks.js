export const heroMock = {
    "id": 13,
    "name": "Chewbacca",
    "films": [
        {id: "1", title: 'A New Hope'},
        {id: "2", title: 'The Empire Strikes Back'},
        {id: "3", title: 'Return of the Jedi'},
        {id: "6", title: '"Revenge of the Sith'},
    ],
   
    "starships": [
        {id: "10", name: 'Millennium Falcon'},
        {id: "22", name: 'Imperial shuttle'},
    ],
    
    "url": "https://sw-api.starnavi.io/people/13/"
}