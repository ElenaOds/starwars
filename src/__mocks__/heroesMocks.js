export const heroesMocks = [
    {
        "id": 13,
        "name": "Chewbacca",
        "height": "228",
        "mass": "112",
        "hair_color": "brown",
        "skin_color": "unknown",
        "eye_color": "blue",
        "birth_year": "200BBY",
        "gender": "male",
        "homeworld": 14,
        "films": [
            1,
            2,
            3,
            6
        ],
        "species": [
            3
        ],
        "vehicles": [
            19
        ],
        "starships": [
            10,
            22
        ],
        "created": "2014-12-10T16:42:45.066000Z",
        "edited": "2014-12-20T21:17:50.332000Z",
        "url": "https://sw-api.starnavi.io/people/13/"
    },

    {
        "id": 34,
        "name": "Finis Valorum",
        "height": "170",
        "mass": "unknown",
        "hair_color": "blond",
        "skin_color": "fair",
        "eye_color": "blue",
        "birth_year": "91BBY",
        "gender": "male",
        "homeworld": 9,
        "films": [
            4
        ],
        "species": [
            1
        ],
        "vehicles": [],
        "starships": [],
        "created": "2014-12-19T17:21:45.915000Z",
        "edited": "2014-12-20T21:17:50.379000Z",
        "url": "https://sw-api.starnavi.io/people/34/"
    },
]