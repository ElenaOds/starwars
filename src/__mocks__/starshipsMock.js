export const starshipsMock = [
    {id: "10", name: 'Millennium Falcon'},
    {id: "22", name: 'Imperial shuttle'},
]
