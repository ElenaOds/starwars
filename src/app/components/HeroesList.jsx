'use client'
import { useEffect, useState } from 'react';

import { getHeroes } from '../../lib/api';
import HeroesListItem from './HeroesListItem';
import Pagination from './Pagination';

export default function HeroesList() {
 const [items, setItems ] = useState([]);
 const [page, setPage] = useState(1);
 const [count, setCount] = useState();

     useEffect(() => {
        const fetchData = async () => {
            try {
              const response = await getHeroes({page});
              setCount(response.count)
              setItems([...response.results]);
            } catch (error) {
              console.error(error);
            }
          };
        fetchData();
      }, [page]);


const handlePreviousPage = () => {
    if (page > 1) {
        setPage((prevPage) => prevPage - 1);
    }
};

const handleNextPage = () => {
    setPage((prevPage) => prevPage + 1);
}


    return (
      <div className="flex flex-col w-full">
      <ul className="grid gap-4 mb-8 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-5">
       {items.map((item) => {
        return (
            <HeroesListItem key={item.id} item={item} />
         )
       })}  
      </ul>
        <Pagination page={page} handlePreviousPage={handlePreviousPage} handleNextPage={handleNextPage} count={count}/>  
      </div>
    );
  } 
  