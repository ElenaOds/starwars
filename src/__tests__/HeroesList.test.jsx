import '@testing-library/jest-dom'
import { render } from '@testing-library/react';

import { heroesMocks } from '../__mocks__/heroesMocks';
import HeroesList from '../app/components/HeroesList';

describe('HeroesList', () => {
    it('renders a list of heroes', () => {

        const { asFragment } = render(<HeroesList heroes={heroesMocks} />);

        expect(asFragment()).toMatchSnapshot();
      });
    });

  
