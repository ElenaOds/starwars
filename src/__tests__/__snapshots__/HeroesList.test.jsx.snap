// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroesList renders a list of heroes 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col w-full"
  >
    <ul
      class="grid gap-4 mb-8 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-5"
    />
    <div
      class="flex gap-4 self-center"
    >
      <button
        class="bg-blue-500 w-20 p-1 rounded-md text-white hover:bg-blue-400 disabled:cursor-not-allowed disabled:bg-blue-200"
        disabled=""
        type="button"
      >
        Previous
      </button>
      <button
        class="bg-blue-500 w-20 p-1 rounded-md text-white hover:bg-blue-400 disabled:cursor-not-allowed disabled:bg-blue-200"
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</DocumentFragment>
`;
